eyeshade/workers/suggestions.js

```diff
@@ -20,43 +20,53 @@ module.exports = (runtime) => {
         continue
       }
 
-      const publisher = suggestion.channel
-      for (let j = 0; j < suggestion.funding.length; j += 1) {
-        const source = suggestion.funding[j]
-        // FIXME
-        const voteValue = '0.25'
-
-        const surveyorId = date + '_' + source.promotion // abuse promotion id as surveyor id
-
-        const surveyorUpdate = `
-        insert into surveyor_groups (id, price, virtual) values ($1, $2, true)
-        on conflict (id) do nothing;
-        `
-        await runtime.postgres.query(surveyorUpdate, [
-          surveyorId,
-          voteValue
-        ], client)
-
-        const cohort = source.type
-
-        const voteUpdate = `
-        insert into votes (id, cohort, tally, excluded, channel, surveyor_id) values ($1, $2, $3, $4, $5, $6)
-        on conflict (id) do update set updated_at = current_timestamp, tally = votes.tally + $3;
-        `
-
-        // This is due to a payout event which occurred early Apr 2022
-        const regex = /.*08277a30-78fd-48a7-a41a-a64b094a2f40.*/g
-        const tally = regex.test(surveyorId) ? '1' : new BigNumber(source.amount).dividedBy(voteValue).toString()
-
-        await runtime.postgres.query(voteUpdate, [
-          votesId(publisher, cohort, surveyorId),
-          cohort,
-          tally,
-          runtime.config.testingCohorts.includes(cohort),
-          publisher,
-          surveyorId
-        ], client)
-      }
+      await handleMessage(runtime, suggestion, client)
     }
   })
 }
+
+async function handleMessage (runtime, suggestion, client, hasValidCountryFunc = hasValidCountry) {
+  const publisher = suggestion.channel
+  const date = moment().format('YYYY-MM-DD')
+
+  if (await hasValidCountryFunc(runtime, publisher)) {
```

Comment:
so, for every message we get on the consumer we will be calling the publishers api to validate if the channel belongs to an allowable country?  Will this not back up the consumers and increase the load a lot to publishers api?  Curious about the impact of doing this and what we will need to scale up.

Reply:
Yes it will increase the load on publishers, and we can work on a more elegant, cached list that runs nightly as a v2. If the consumers slow down too much, we can partition. 

We could also do it during payout, but that doesn't quite meet the requirements. 

If publishers is down, for any Exception, we let the vote through by default.